src/Test/Hspec/Expectations.hs

```diff
@@ -87,6 +88,24 @@ infix 1 `shouldNotBe`, `shouldNotSatisfy`, `shouldNotContain`, `shouldNotReturn`
 with_loc(shouldBe, (Show a, Eq a) => a -> a -> Expectation)
 actual `shouldBe` expected = expectTrue ("expected: " ++ show expected ++ "\n but got: " ++ show actual) (actual == expected)
 
+-- |
+-- @actual \`shouldBeNear\` expected@ sets the expectation that @actual@ be a
+-- floating point value near @expected@. If either value is zero we check that
+-- the absolute difference is less than epsilon (1e-15) otherwise we check if
+-- the relative difference is less than epsilon.
+with_loc(shouldBeNear, (Show a, Ord a, Floating a) => a -> a -> Expectation)
+actual `shouldBeNear` expected
+  -- Short circuit if they are actually equal.
+  | actual == expected = reportFail (actual == expected)
+  | actual == 0 || expected == 0 = reportFail (absoluteDifference < epsilon)
+  | otherwise =  reportFail (relativeDifference < epsilon)
+  where
+    epsilon = 1e-15
+    absoluteDifference = abs (expected - actual)
+    relativeDifference = absoluteDifference / (abs actual + abs expected)
+    reportFail =
+      expectTrue ("expected: " ++ show expected ++ "\n but got: " ++ show actual)
```

Comment:
+1 on this change BTW!

Wouldn't it be nicer to report things like `expected 123 ± 1% but got 125` or similar, as the expectation is of it being within a tolerance of the value not _being_ the value?